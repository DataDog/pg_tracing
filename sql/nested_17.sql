-- Test error thrown within a nested function
/*traceparent='00-00000000000000000000000000000001-0000000000000001-01'*/ SELECT
    eval_expr(format('date %L + interval %L', '-infinity', 'infinity')),
    eval_expr(format('date %L - interval %L', '-infinity', 'infinity'));
select span_operation, lvl FROM peek_ordered_spans where trace_id='00000000000000000000000000000001';

-- Cleanup
CALL clean_spans();
