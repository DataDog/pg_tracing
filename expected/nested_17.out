-- Test error thrown within a nested function
/*traceparent='00-00000000000000000000000000000001-0000000000000001-01'*/ SELECT
    eval_expr(format('date %L + interval %L', '-infinity', 'infinity')),
    eval_expr(format('date %L - interval %L', '-infinity', 'infinity'));
       eval_expr        | eval_expr 
------------------------+-----------
 timestamp out of range | -infinity
(1 row)

select span_operation, lvl FROM peek_ordered_spans where trace_id='00000000000000000000000000000001';
                            span_operation                            | lvl 
----------------------------------------------------------------------+-----
 SELECT eval_expr(format($1, $2, $3)), eval_expr(format($4, $5, $6)); |   1
 Planner                                                              |   2
 ExecutorRun                                                          |   2
 Result                                                               |   3
 $2||expr                                                             |   4
 Planner                                                              |   5
 Planner                                                              |   5
 select date $1 - interval $2                                         |   4
 Planner                                                              |   5
 ExecutorRun                                                          |   5
 Result                                                               |   6
(11 rows)

-- Cleanup
CALL clean_spans();
